shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -927,4 +885,67 @@ void AccessibilityBridge::FillInspectTree(int32_t flutter_node_id,
 }
 #endif  // !FLUTTER_RELEASE
 
+AccessibilityBridge::FuchsiaAtomicUpdate::FuchsiaAtomicUpdate(
+    std::queue<FuchsiaAtomicUpdate>* atomic_updates)
+    : atomic_updates_(atomic_updates) {}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeUpdate(
+    fuchsia::accessibility::semantics::Node node,
+    size_t size) {
+  if (size > kMaxMessageSize) {
+    // TODO(MI4-2531, FIDL-718): Remove this
+    // This is defensive. If, despite our best efforts, we ended up with a node
+    // that is larger than the max fidl size, we send no updates.
+    PrintNodeSizeError(node.node_id());
+    return;
+  }
+  updates_.emplace_back(std::move(node), size);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeDeletion(uint32_t id) {
+  deletions_.push_back(id);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::Apply(
+    fuchsia::accessibility::semantics::SemanticTreePtr* tree) {
+  const size_t kMaxDeletionsPerUpdate = kMaxMessageSize / kNodeIdSize;
+  size_t begin = 0;
+  auto it = deletions_.begin();
+  while (it != deletions_.end()) {
+    std::vector<uint32_t> to_delete;
+    size_t end = std::min(deletions_.size() - begin, kMaxDeletionsPerUpdate);
+    std::copy(std::make_move_iterator(it), std::make_move_iterator(it + end),
+              std::back_inserter(to_delete));
+    (*tree)->DeleteSemanticNodes(std::move(to_delete));
+    begin = end;
+    it += end;
+  }
+
+  std::vector<fuchsia::accessibility::semantics::Node> to_update;
+  size_t current_size = 0;
+  for (auto& node_and_size : updates_) {
+    if (current_size + node_and_size.second > kMaxMessageSize) {
+      (*tree)->UpdateSemanticNodes(std::move(to_update));
+      current_size = 0;
+      to_update.clear();
+    }
+    current_size += node_and_size.second;
+    to_update.push_back(std::move(node_and_size.first));
+  }
+
+  if (!to_update.empty()) {
```

Comment:
Sorry David I am having a hard time to understand what you meant, maybe because the way the diff is presented to me is different the way you selected the lines. Can you paste exactly on a comment what you mean?